components/admin/users/AdminUsersSplitSearch.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import _ from 'lodash'
+
+const split: any = (
+  str: string,
+  searchTerm: string,
+  strIndex: number = 0,
+  res: string[] = [],
+  current: string = ''
+) => {
+  if (strIndex === str.length) {
+    current && res.push(current)
+    return res
+  }
+
+  if (str[strIndex] === searchTerm[current.length]) {
+    /*
+    single character in searchTerm has been found
+    Ex:
+      searchTerm = 'bonjour'
+      current = 'bon' <--- previous characters
+      str[strIndex] = 'j' <--- current character
+    */
+    current += str[strIndex]
+
+    //searchTerm has been fully found, push to array and reset current
+    if (current.length === searchTerm.length) {
+      res.push(current)
+      current = ''
+    }
+  } else if (current.length) {
+    /*
+    Character in searchTerm was previously found, but current character does not match
+    Ex:
+      searchTerm = 'bonjour'
+      current = 'bon' <--- previous characters
+      str[strIndex] = 'z' <--- current character
+    */
+    res.push(current)
+
+    //check if current character is first character of searchTerm
+    current = str[strIndex] === searchTerm[0] ? str[strIndex] : ''
+
+    !current && res.push(str[strIndex])
+  } else {
+    res.push(str[strIndex])
+  }
+
+  return split(str, searchTerm, strIndex + 1, res, current)
+}
+
+export const AdminUsersSplitSearch: any = (str: string, searchTerm: string) => {
+  const splitArr = split(str, searchTerm)
+  const value = splitArr.map((word: string) => {
+    const bgColor = word !== searchTerm ? 'none' : 'rgb(84, 64, 216, .25)'
+    return (
+      <span style={{ backgroundColor: bgColor }} key={_.uniqueId()}>
```

Comment:
I thought `key={_.uniqueId()}` is bad practice. it forces constant refreshes even when unnecessary.

Reply:
It is needed in this instance, or else the `highlighting of searchTerm` will not highlight the correct words.

Example: see the last row  in this column. It is highlighted incorrectly when not using `key={_.uniqueId()}`
<img width="1116" alt="Screen Shot 2020-08-10 at 10 52 48 AM" src="https://user-images.githubusercontent.com/45890848/89814462-03e05700-daf8-11ea-928f-33c555ed735f.png">

Reply:
then the key should be `word+searchTerm+index`. This way, if the search term and the word stays the same, it will not rerender.